src/integration/java/io/pinecone/helpers/IndexManager.java

```diff
@@ -79,6 +84,76 @@ private static String createNewIndex(PineconeControlPlaneClient controlPlaneClie
         return indexName;
     }
 
+    public static IndexModel waitUntilIndexIsReady(PineconeControlPlaneClient controlPlaneClient, String indexName, Integer totalMsToWait) throws InterruptedException {
+        IndexModel index = controlPlaneClient.describeIndex(indexName);
+        int waitedTimeMs = 0;
+        int intervalMs = 1500;
+
+        while (!index.getStatus().getReady()) {
+            index = controlPlaneClient.describeIndex(indexName);
+            if (waitedTimeMs >= totalMsToWait) {
+                System.out.println("Index " + indexName + " not ready after " + waitedTimeMs + "ms");
```

Comment:
Let's replace this with logger.info

Reply:
Example: 
Define this at the class level:
private static final Logger logger = LoggerFactory.getLogger(IndexManager.class);

Within the method where we need to log:
logger.info("logging statement");

Reply:
Good point, let me take a look at updating all the println statements.